sdk/search/Azure.Search.Documents/tests/Utilities/SearchRecordedTestSanitizer.cs

```diff
@@ -19,34 +20,51 @@ public class SearchRecordedTestSanitizer : RecordedTestSanitizer
         /// </summary>
         private const string ApiKeyHeaderName = "api-key";
 
-        /// <summary>
-        /// Secret values to sanitize from body.
-        /// </summary>
-        private readonly HashSet<string> _secrets = new HashSet<string>();
+        public override void Sanitize(RecordSession session)
+        {
+            var secrets = new HashSet<string>();
 
-        /// <summary>
-        /// Redact sensitive body content.
-        /// </summary>
-        /// <param name="contentType">The Content-Type of the body content.</param>
-        /// <param name="body">The body content.</param>
-        /// <returns>The sanitized body content.</returns>
-        public override string SanitizeTextBody(string contentType, string body)
+            foreach (var variable in session.Variables.ToArray())
+            {
+                if (SearchTestEnvironment.StorageAccountKeyVariableName.Equals(variable.Key, StringComparison.OrdinalIgnoreCase))
+                {
+                    // Assumes the secret content is destined to appear in JSON, for which certain common characters in account keys are escaped.
+                    // See https://github.com/dotnet/runtime/blob/8640eed0/src/libraries/System.Text.Json/src/System/Text/Json/Writer/JsonWriterHelper.Escaping.cs
+                    string encoded = JavaScriptEncoder.Default.Encode(variable.Value);
+                    secrets.Add(encoded);
+
+                    session.Variables[variable.Key] = SanitizeValue;
+                }
+            }
+
+            if (secrets.Count > 0)
+            {
+                foreach (var recordEntry in session.Entries)
+                {
+                    SanitizeSecretsInJsonBody(secrets, recordEntry.Request);
+                    SanitizeSecretsInJsonBody(secrets, recordEntry.Response);
+                }
+            }
+
+            base.Sanitize(session);
+        }
+
+        internal void SanitizeSecretsInJsonBody(HashSet<string> secrets, RecordEntryMessage message)
         {
-            if (_secrets.Count > 0 &&
+            if (message.TryGetBodyAsText(out var body) &&
+                message.TryGetContentType(out var contentType) &&
                 !string.IsNullOrEmpty(body) &&
                 contentType != null &&
                 contentType.StartsWith("application/json", StringComparison.OrdinalIgnoreCase))
             {
                 StringBuilder sanitized = new StringBuilder(body);
-                foreach (string secret in _secrets)
+                foreach (string secret in secrets)
                 {
                     sanitized.Replace(secret, SanitizeValue);
                 }
 
-                return sanitized.ToString();
+                message.Body = Encoding.UTF8.GetBytes(sanitized.ToString());
```

Comment:
You also have to call `UpdateSanitizedContentLength` - duplicating everything `SanitizeBody` does already.

Reply:
You don't base class would call it for you.